indexing-service/src/main/java/io/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -164,12 +165,20 @@ public Firehose connect(InputRowParser inputRowParser) throws IOException, Parse
       List<String> dims;
       if (dimensions != null) {
         dims = dimensions;
+      } else if (inputRowParser.getParseSpec().getDimensionsSpec().hasCustomDimensions()) {
+        dims = inputRowParser.getParseSpec().getDimensionsSpec().getDimensions();
```

Comment:
since dimensions and dimensions exclusions cannot overlap, so no need to check dimensionExclusions if dimensions are explicitly specified.